sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -40,26 +40,39 @@ std::unique_ptr<RawResponse> TransportPolicy::Send(
   (void)nextHttpPolicy;
   ctx.ThrowIfCancelled();
 
-  /**
+  /*
    * The transport policy is always the last policy.
-   * Call the transport and return
+   *
+   * Default behavior for all requests is to download the full response to the RawResponse's
+   * buffer.
+   *
+   ********************************** Notes ************************************************
+   *
+   * - If ReadToEnd() fails while downloading all the response, the retry policy will make sure to
+   * re-send the request to re-start the download.
+   *
+   * - If the request returns error (statusCode < 300), even if `!request.IsBufferedDownload()`, the
+   * response will be download to the response's buffer.
```

Comment:
Can you help me understand why we do this, for error codes < 300?

And don't you mean >= 300? Basically, for errors, we always have response in a buffer, and not in a stream, correct?

```suggestion
   * - If the request returns error (statusCode >= 300), even if `request.IsBufferedDownload()`, the
   * response will be download to the response's buffer.
```

Reply:
Say we want to download a Blob. We call `auto blobResponse = client.Download(...)`.
We are expecting the blobResponse to contain a bodyStream that we can use to read bytes from the network, but, somehow, client.Download() fail and throw an exception, and we were prepared by catching RequestException const& exception.
For that `exception`, we won't expect a bodyStream. All the error payload can be downloaded and delivered to the caller wrapped in the exception.

Well, that's what we are doing here. We only honor returning the bodyStream for successful requests. Otherwise, we forget about the request has asked for a bodyStream and we just return a body buffer.  If we weren't, and always honor returning the body stream, the caller (the service's developer) would check the response has failed with error code and pull the entire response from the stream anyway... So, that's what we are doing for them. When the caller checks the raw response and sees the error code, they are know they can take the response already from the buffer in the response because the awesome transport adapter went beyond expectation and knew that's was going to be required :)